components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    aes_gcm_decrypt, aes_gcm_encrypt,
+    error::{JwCryptoError, Result},
+    Algorithm, CompactJwe, EncryptionAlgorithm, JweHeader, Jwk, JwkKeyParameters,
+};
+
+pub(crate) fn encrypt_to_jwe(data: &[u8], jwk: &Jwk) -> Result<CompactJwe> {
+    let protected_header = JweHeader {
+        kid: jwk.kid.clone(),
+        alg: Algorithm::Direct,
+        enc: EncryptionAlgorithm::A256GCM, // ??
+        epk: None,
+        apu: None,
+        apv: None,
```

Comment:
It's unfortunate that you have to specify these ECDH-specific params as `None` here, but I can imagine it would be a bit of a refactor to move them into a ECDH-specific sub-struct or whatever.